knime_extension/src/nodes/spatialstatistics.py

```diff
@@ -1044,3 +1067,240 @@ def execute(self, exec_context: knext.ExecutionContext, input_1, input_2):
         plt.title("Local Getis-Ord G Scatterplot")
 
         return knext.Table.from_pandas(gdf), knext.view_matplotlib(f)
+
+
+# new added nodes: 2023-11-27
+
+
+############################################
+# Bivariate Global Moran’s I node
+############################################
+# FIXME: add test workflow
+@knext.node(
+    name="Bivariate Global Moran’s I",
+    node_type=knext.NodeType.LEARNER,
+    # node_type=knext.NodeType.MANIPULATOR,
+    category=__category,
+    icon_path=__NODE_ICON_PATH + "BivariateGlobal.png",
+)
+@knext.input_table(
+    name="Input Table",
+    description="Input table for calculation of Bivariate Global Moran’s I",
+)
+@knext.input_table(
+    name="Spatial Weights",
+    description="Spatial Weights table for calculation of Bivariate Global Moran’s I",
+)
+@knext.output_table(
+    name="Output Table",
+    description="Output table results of Bivariate Global Moran’s I"
+    + __global_statistics_output_table_description,
+)
+# @knext.output_binary(
+#     name="output model",
+#     description="Output model of Bivariate Global Moran’s I",
+#     id="pysal.esda.moran.Moran",
+# )
+@knext.output_view(
+    name="output view",
+    description="Output view of Bivariate Global Moran’s I"
+    + __global_statistics_interactive_view_description,
+)
+class BivariateGlobalMoran:
+    """
+    Bivariate Global Moran’s I.
+    """
+
+    # input parameters
+    geo_col = knext.ColumnParameter(
+        "Geometry column",
+        "The column containing the geometry to use for Bivariate Global Moran’s I.",
+        column_filter=knut.is_geo,
+        include_row_key=False,
+        include_none_column=False,
+    )
+
+    id_col = get_id_col_parameter()
+
+    Field_col1 = knext.ColumnParameter(
```

Comment:
Please change the case of this field and all following to be consistent. If you have existing workflows just search in the corresponding node settings and replace the old with the new parameter name.